src/client.ts

```diff
@@ -5245,14 +5245,15 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * <p>If the EventTimelineSet object already has the given event in its store, the
      * corresponding timeline will be returned. Otherwise, a /context request is
      * made, and used to construct an EventTimeline.
+     * If the event does not belong to this EventTimelineSet then undefined will be returned.
      *
-     * @param {EventTimelineSet} timelineSet  The timelineSet to look for the event in
+     * @param {EventTimelineSet} timelineSet  The timelineSet to look for the event in, must be bound to a room
```

Comment:
We should add an assertion or throw a useful error when `timelineSet.room` is not set. Currently it will blow up with a `undefined` exception.

Reply:
Done in https://github.com/matrix-org/matrix-js-sdk/pull/2454#discussion_r896279149 ✅